modules/datastore/src/Form/DashboardForm.php

```diff
@@ -0,0 +1,440 @@
+   *
+   * @return array[]
+   *   Table render array.
+   */
+  public function buildTable(array $datasets): array {
+    return [
+      'table' => [
+        '#theme' => 'table',
+        '#weight' => 3,
+        '#header' => self::DATASET_HEADERS,
+        '#rows' => $this->buildDatasetRows($datasets),
+        '#attributes' => ['class' => 'dashboard-datasets'],
+        '#attached' => ['library' => ['harvest/style']],
+        '#empty' => 'No datasets found',
+      ],
+      'pager' => [
+        '#type' => 'pager',
+      ],
+    ];
+  }
+
+  /**
+   * Retrieve list of UUIDs for datasets matching the given filters.
+   *
+   * @param string[] $filters
+   *   Datasets filters.
+   *
+   * @return string[]
+   *   Filtered list of dataset UUIDs.
+   */
+  protected function getDatasets(array $filters): array {
+    $datasets = [];
+
+    // If a value was supplied for the UUID filter, include only it in the list
+    // of dataset UUIDs returned.
+    if (isset($filters['uuid'])) {
+      $datasets = [$filters['uuid']];
+    }
+    // If a value was supplied for the harvest ID filter, retrieve dataset UUIDs
+    // belonging to the specfied harvest.
+    elseif (isset($filters['harvest_id'])) {
+      $harvestLoad = $this->getHarvestLoadStatus($filters['harvest_id']);
+      $datasets = array_keys($harvestLoad);
+      $total = count($datasets);
+      $currentPage = $this->pagerManager->createPager($total, $this->itemsPerPage)->getCurrentPage();
+
+      $chunks = array_chunk($datasets, $this->itemsPerPage) ?: [[]];
+      $datasets = $chunks[$currentPage];
+    }
+    // If no filter values were supplied, fetch from the list of all dataset
+    // UUIDs.
+    else {
+      $total = $this->metastore->count('dataset');
+      $currentPage = $this->pagerManager->createPager($total, $this->itemsPerPage)->getCurrentPage();
+      $datasets = $this->metastore->getRangeUuids('dataset', $currentPage, $this->itemsPerPage);
+    }
+
+    return $datasets;
+  }
+
+  /**
+   * Builds dataset rows array.
+   *
+   * @param string[] $datasets
+   *   Dataset UUIDs for which to generate dataset rows.
+   *
+   * @return array
+   *   Table rows.
+   */
+  protected function buildDatasetRows(array $datasets): array {
+    // Fetch the status of all harvests.
+    $harvestLoad = iterator_to_array($this->getHarvestLoadStatuses());
+
+    $rows = [];
+    // Build dataset rows fore each of the supplied dataset UUIDs.
+    foreach ($datasets as $datasetId) {
+      // Gather dataset information.
+      $datasetInfo = $this->datasetInfo->gather($datasetId);
+      if (empty($datasetInfo['latest_revision'])) {
+        continue;
+      }
+      // Build a table row using it's details and harvest status.
+      $datasetRow = $this->buildDatasetRow($datasetInfo, $harvestLoad[$datasetId] ?? 'N/A');
+      $rows = array_merge($rows, $datasetRow);
+    }
+
+    return $rows;
+  }
+
+  /**
+   * Fetch the status of all harvests.
+   *
+   * @return \Generator
```

Comment:
@clayliddell idk if upgrading codeclimate will fix this, but really this is not a very useful piece of documentation anyway - it's technically correct but doesn't really tell you what is being yielded. I'd say just remove this line.

Reply:
Fair enough. Resolved in https://github.com/GetDKAN/dkan/pull/3714#discussion_r760346155.